File: Draft_code/nodet/nodet.cpp
#include <iostream>
using namespace std;

template<class T>
class NodeT {
    public:
    T data;
    NodeT *next;
    NodeT(T data) {
        this->data = data;
        this->next = nullptr;
    }
    T get_data()
    {
        return this->data;
    }
    void set_data(T data)
    {
        this->data = data;
    }
};

void run_node() {
    NodeT<int> node = NodeT<int>(10);
    cout << node.get_data() << endl;
    node.set_data(20);
    cout << node.get_data() << endl;
}

int main() {
    run_node();
    return 0;
}

File: Draft_code/nodet/nodet.ts
// Typescript recommend to write member functions inside the class
class NodeT<T> {
  data: T;
  next: NodeT<T> | null;
  
  constructor(data: T) {
    this.data = data;
    this.next = null;
  }

  get_data(): T {
    return this.data;
  }

  set_data(data: T): void {
    this.data = data;
  }
}

function run_node() {
    var node = new NodeT(10);
    console.log(node.get_data());
    node.set_data(20);
    console.log(node.get_data());
}

// main part
run_node();

